src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+        if a_name != self._experiment_name:
+            self._experiment_name = a_name
+            self.light.load_experiment(self._experiment_name)
+
+    @property
+    def grating(self):
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, grating_string):
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        if grating_string in self.gratings_options:
+            self.light.set(lf.AddIns.SpectrometerSettings.GratingSelected, grating_string)
+        else:
+            logger.error(f"Grating {grating_string} is not an options. The options are: {self.gratings_options}")
+
+    @property
+    def gratings_options(self):
+        """
+        Returns a list of all installed gratings.
+        """
+        #NOTE: This line below is important. "GetCurrentCapabilities" is able to return the list of possibilities of any Lightfield call in order to provide more information.
+        available_gratings = self.light.experiment.GetCurrentCapabilities(lf.AddIns.SpectrometerSettings.GratingSelected)
+        return [str(a) for a in available_gratings]
+
+    @property
+    def num_frames(self):
+        """
+        Returns the number of frames taken during the acquisition.
+        """
+        return self.light.get(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore)
+
+    @num_frames.setter
+    def num_frames(self, num_frames):
+        """
+        Sets the number of frames to be taken during acquisition to number.
+        """
+        return self.light.set(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore, num_frames)
+
+    @property
+    def exposure_time(self):
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms):
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        return self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, ms)
+
+    @property
+    def sensor_temperature(self):
+        """
+        Returns the current sensor temperature (in celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureReading)
+
+    @property
+    def temperature_sensor_setpoint(self):
+        """
+        Returns the sensor setpoint temperature (celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+
+    @temperature_sensor_setpoint.setter
+    def temperature_sensor_setpoint(self, deg_C):
+        """
+        Sets the sensor target temperature (in degrees Celsius) to deg_C.
+        This function retrieves image data, with particular attention to the camera's configuration determined by the `temperature_sensor_setpoint`.
```

Comment:
Does this function return data? That's confusing for a setter.

Reply:
I think you're one of the experts of the spectrometer now. If you could return links to find more information about these settings, that could be super helpful.

Reply:
Fixed